### ISSUE TITLE
Source path directory too long

### ISSUE THREAD

**Issue by addisonElliott** [InsightSoftwareConsortium/ITKPythonPackage#67]
When attempting to install via the pip:
`pip install itk`

I receive the following error:
```
ITK source code directory path length is too long (${n} > 50).
Please move the ITK source code directory to a directory with a shorter path.
```

According to this article (https://public.kitware.com/pipermail/community/2013-December/004475.html), the check is due to Visual Studio and truncation of commands.

Is there a way to specify the build and install directory? In addition, the build files in skbuild are fairly long as it is so I could **only** place the ITKPythonPackage directory in my D: drive for it to work.

If there is not a way to specify the build and install directory, I propose this option be added and documentation be added to the front page.

**Comment by fbudin69500** [comment]
Thanks for reporting this issue. Could you provide more information about your system?
Which Windows, Visual Studio, and Python versions are you using? I am surprised that pip does not download automatically the binary wheel available on PyPI.

PS: A good place to start a conversion about a problem you are having with ITK is [our discourse forum](https://discourse.itk.org/).

**Comment by thewtex** [comment]
Hi @addisonElliott,

Before running

```
pip install itk
```

was

```
pip install --upgrade pip
```

executed? Some older versions of pip will not find the binaries correctly.

**Comment by addisonElliott** [comment]
Thanks for the comments guys.

@fbudin69500 I am running Windows 10 64-bit, Python 3.5.2, Microsoft Visual Studio 14 (2015). I found out the reason why a precompiled wheel was not being installed is because I was using a 32-bit version of Python (no idea why I downloaded 32-bit, probably just didnt notice).

@thewtex Yes, I tried that multiple times just in case and no luck.

I was able to install ITK by installing a 64-bit version of Python and then running the two commands. However, the issue still stands that installing itk from source on Windows will likely fail and the user will be left to not knowing why.

I found that you **briefly** discussed this in your docs here:
> We need to work in a short directory to avoid path length limitations on Windows, so the repository is cloned into C:IPP.

Source: http://itkpythonpackage.readthedocs.io/en/latest/Build_ITK_Python_packages.html

I recommend updating the README and Quick start guide to include this tidbit.

Here is what I think should be added:
If a precompiled wheel file is not found for the ITK package, then it will attempt to build from source. On Windows machines, the source path cannot be greater than 50 characters or it will cause issues down the line due to Visual Studio truncating filenames. If you must compile from source, clone this repository in a short directory like C:/IPP. Then, run setup.py within the repository via command line.

If I get some time, I'll try to submit a pull request. However, if anyone else has the time, then feel free to implement the changes yourselves.
